front/components/assistant_builder/Sharing.tsx

```diff
@@ -0,0 +1,327 @@
+import {
+  ChevronDownIcon,
+  Chip,
+  Dialog,
+  DropdownMenu,
+  DustIcon,
+  IconButton,
+  LockIcon,
+  PlanetIcon,
+  UserGroupIcon,
+} from "@dust-tt/sparkle";
+import type { AgentConfigurationScope, WorkspaceType } from "@dust-tt/types";
+import { isBuilder } from "@dust-tt/types";
+import { useState } from "react";
+
+import { assistantUsageMessage } from "@app/lib/assistant";
+import { useAgentConfiguration, useAgentUsage } from "@app/lib/swr";
+
+type ConfirmationModalDataType = {
+  title: string;
+  text: string;
+  confirmText: string;
+  showUsage?: boolean;
+  variant: "primary" | "primaryWarning";
+};
+
+const SCOPE_INFO: Record<
```

Comment:
I'll move this to a dedicated component so we can avoid duplicating this logic everywhere.

Reply:
Agreed there should be an `export` here
IMHO this is the dedicated lib of components (Sharing); when it becomes too large we could indeed split into front & lib components

Reply:
Seen IRL, this comment was meant for the Chip logic that will used both in the AssistantPreview and the assistant builder logic.

Reply:
As discussed will extract a SharingChip & move Sharing to another place